src/main/java/com/dbteam/service/serviceImpl/PaymentServiceImpl.java

```diff
@@ -68,4 +68,20 @@ public void addPayment(Payment payment) {
         return paymentRepository
                 .getPaymentsByGroupChatIdAndIsConfirmedIsFalse(groupChatId);
     }
+
+    @Override
+    public Payment getFirstPaymentWithUserBefore(LocalDate date, String username) {
+        Optional<Payment> payment = paymentRepository
+                .getPaymentByRecipientEqualsOrPayerEqualsAndDateBeforeOrderByDateDesc(
+                        username, username, date).findFirst();
+        return payment.orElse(null);
+    }
+
+    @Override
+    public Payment getFirstPaymentWithUserAfter(LocalDate date, String username) {
+        Optional<Payment> payment = paymentRepository
+                .getPaymentByRecipientEqualsOrPayerEqualsAndDateAfterOrderByDateDesc(
+                        username, username, date).findFirst();
+        return payment.orElse(null);
```

Comment:
IMO, Repository should return `Optional<Payment>` and here we'll have:
`return paymentRepository.getPaymentByRecipientEqualsOrPayerEqualsAndDateAfterOrderByDateDesc(username, username, date).orElse(something)`

Reply:
The idea is to make it as easy as possible to change in future if we would like more than one payment. Then we'd have to change only Service and not also repository. Streams are great in this sense because of the reasons stated here:

https://www.baeldung.com/spring-data-java-8

But streams are slower than just one Object, so your remark is right. So we'll do that.